Extensions/DraggableBehavior/draggableruntimebehavior.ts

```diff
@@ -172,15 +180,26 @@ namespace gdjs {
       runtimeScene: gdjs.RuntimeScene,
       draggableRuntimeBehavior: DraggableRuntimeBehavior
     ) {
+      if (
+        this._draggableBehavior &&
+        draggableRuntimeBehavior.owner.getZOrder() <=
+          this._draggableBehavior.owner.getZOrder()
+      ) {
+        return false;
+      }
       const position = this.getPosition(runtimeScene, draggableRuntimeBehavior);
       if (
         !draggableRuntimeBehavior.owner.insideObject(position[0], position[1])
       ) {
         return false;
       }
+      if (this._draggableBehavior) {
+        // The previous best object to drag will not be dragged.
+        this._draggableBehavior._draggedByDraggableManager = null;
```

Comment:
nitpicking: I think we could make this (even if it's in a simple line) into a `dismissDrag`? Might help making it clear that we're not ending the dragging, we're dismissing it before it was even started.